lib/mixlib/install/backend/artifactory.rb

```diff
@@ -0,0 +1,70 @@
+#
+# Author:: Patrick Wright (<patrick@chef.io>)
+# Copyright:: Copyright (c) 2015 Chef, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "net/http"
+require "json"
+require "mixlib/install/artifact_info"
+require "artifactory"
+
+module Mixlib
+  class Install
+    class Backend
+      class Artifactory
+        ARTIFACTORY_ENDPOINT = "http://artifactory.chef.co".freeze
+
+        attr_accessor :options
+
+        def initialize(options)
+          @options = options
+          @client = ::Artifactory::Client.new(endpoint: ARTIFACTORY_ENDPOINT)
+        end
+
+        def info
+          params = {
+            "repos" => "omnibus-current-local",
+            "omnibus.platform" => options.platform,
+            "omnibus.platform_version" => options.platform_version,
+            "omnibus.architecture" => options.architecture,
+            "omnibus.version" => options.product_version
+          }
+
+          headers = { "X-Result-Detail" => "properties" }
+
+          begin
+            artifact = @client.get("/api/search/prop", params, headers)
```

Comment:
My two cents worth:

What are you really trying to test? Are you trying to test that the endpoint itself is functioning; or the upstream artifactory gem interaction? I would argue that the case should be the latter, and as such the client.get call/response should be stubbed so that the gem interaction can be tested without external dependencies.

Having said this, there is still value in testing the endpoint. However, I am in the opinion done somewhere else, and substantially more frequently than it would be exercised here. An example of such use would be Chef Audit Mode on the boxes which are running Artifactory, as this would provide an immediate alert to an issue from a broken endpoint due to a change. 

That's my two cents though.